### DISCUSSION TITLE
Data augmentation per patch (not per image)

### DISCUSSION THREAD

**Discussion by jmlipman** [Project-MONAI/tutorials#1206] [Q&A]
Hi,

Does MONAI apply data augmentation per image or per patch? I would say that when dealing with small images (say, 300x300x300), it probably makes sense to apply DA for each image. However, when dealing with a few very-large images (say, 3000x3000x300), it probably makes more sense to apply DA per patch as you will read each image only a few times. In case DA is applied per image, do you think it would be easy to modify MONAI to do it patch-wise?

Thanks!

**Comment by KumoLiu** [comment]
Hi @jmlipman, typically, once all non-random transforms have been applied and the data has been cropped to the patches, we perform data augmentation. And if you used `CacheDataset`, it will cache deterministic transforms' result during training.
https://github.com/Project-MONAI/MONAI/blob/82cd668c4dcb23d36beeb432cc0aa04b9e0a1b3c/monai/data/dataset.py#L695

Thanks!
